### PR TITLE
feat: support vite in v14 bootstrap

### LINKED ISSUE #12057: [Vite] Export web components must work
### How does it work today

- An "export" bundle is created with everything used from exported components - this needs to be verified
    - For each WebComponentExporter a .js file is generated into {build.folder}/frontend and added as an import statement to the generated-flow-imports.js

### How could it work

- In a similar way by creating multiple bundles

### PR DESCRIPTION
**Pull request by caalador**
Support vite build in the v14 bootstrap
to have exported webcomponents working.

Closes #12057

### CODE REVIEW

**Comment by Artur-** [comment]
Test?

**Comment by caalador** [comment]
Added tests

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java [hunk 1]
```diff
@@ -876,6 +879,43 @@ private void setupFrameworkLibraries(Element head,
...
+                    head.appendChild(script.attr("type", "module"));
+                    return;
+                }
+
+                // Get the index.html to get vite generated bundles
+                String index = FrontendUtils.getIndexHtmlContent(service);
+
+                // Get and add all javascriptbundles
+                Matcher scriptMatcher = Pattern
+                        .compile("src=\\\"VAADIN\\/build\\/(.*.js)\\\"")
```
Comment (joheriks) [inline flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java:897]:
```suggestion
                        .compile("src=\\\"VAADIN\\/build\\/(.*\\.js)\\\"")
```

File: flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java [hunk 2]
```diff
@@ -876,6 +879,43 @@ private void setupFrameworkLibraries(Element head,
...
+                            "VAADIN/build/" + scriptMatcher.group(1), false);
+                    head.appendChild(
+                            script.attr("type", "module").attr("async", true)
+                                    // Fixes basic auth in Safari #6560
+                                    .attr("crossorigin", true));
+                }
+
+                // Get and add all css bundle links
+                Matcher cssMatcher = Pattern
+                        .compile("href=\\\"VAADIN\\/build\\/(.*.css)\\\"")
```
Comment (joheriks) [inline flow-server/src/main/java/com/vaadin/flow/server/BootstrapHandler.java:910]:
```suggestion
                        .compile("href=\\\"VAADIN\\/build\\/(.*\\.css)\\\"")
```

**Review by joheriks — CHANGES_REQUESTED** [review]

**Review by joheriks — APPROVED** [review]

**Pull request merged**